addons/tracing/tracing-decision-springboot-addon/src/main/java/org/kie/kogito/tracing/decision/SpringBootDecisionTracingConfiguration.java

```diff
@@ -55,6 +58,22 @@ public SpringBootDecisionTracingConfiguration(
         this.kafkaTopicReplicationFactor = kafkaTopicReplicationFactor;
     }
 
+    @Bean
+    @ConditionalOnProperty(value = "kogito.addon.tracing.decision.asyncEnabled", havingValue = "true", matchIfMissing = true)
+    public SpringBootDecisionTracingCollector collectorForAsyncMode(final SpringBootTraceEventEmitter eventEmitter,
+                                                                    final ConfigBean configBean,
+                                                                    final Application application) {
+        return new SpringBootDecisionTracingCollectorAsync(eventEmitter, configBean, application);
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "kogito.addon.tracing.decision.asyncEnabled", havingValue = "false")
+    public SpringBootDecisionTracingCollector collectorForSyncMode(final SpringBootTraceEventEmitter eventEmitter,
+                                                                   final ConfigBean configBean,
+                                                                   final Application application) {
+        return new SpringBootDecisionTracingCollectorAsync(eventEmitter, configBean, application);
+    }
```

Comment:
Note 1: both are returning `SpringBootDecisionTracingCollectorAsync` instance
Note 2: what about adding the property as method parameter to have a single method? In this way we can prevent wrong situations like generate both beans if we will change parameter logic. Then please add a test to verify this producer method
```suggestion
    @Bean
    public SpringBootDecisionTracingCollector collectorForAsyncMode(final SpringBootTraceEventEmitter eventEmitter,
                                                                    final ConfigBean configBean,
                                                                    final Application application,
                                                                    @Value(value = "${kogito.addon.tracing.decision.asyncEnabled:true}") final boolean asyncEnabled) {
        if (asyncEnabled) {
            return new SpringBootDecisionTracingCollectorAsync(eventEmitter, configBean, application);
        } else {
            return new SpringBootDecisionTracingCollector(eventEmitter, configBean, application);
        }
    }
```

Reply:
Thanks for the catch and the suggestion. It's way better now.

Reply:
Can you please add a test for this method?

Reply:
Done :+1: